js/scripts.js

```diff
@@ -1,36 +1,62 @@
-let pokemonList = [
-    {
-        name: "Diglett",
-        height: 0.08,
-        types: ['ground']
-    },
-    {
-        name: "Lickitung",
-        height: 3.11,
-        types: ['normal']
-    },
-    {
-        name: "Farfetch\'d",
-        height: 2.07,
-        types: ['normal', 'flying']
-    },
-    {
-        name: "Jigglypuff",
-        height: 1.08,
-        types: ['normal', 'fairy']
-    },
-    {
-        name: "Oddish",
-        height: 1.08,
-        types: ['grass', 'poison']
+let pokemonRespository = (function() {
+    let pokemonList = [
+        {
+            name: "Diglett",
+            height: 0.08,
+            types: ['ground']
+        },
+        {
+            name: "Lickitung",
+            height: 3.11,
+            types: ['normal']
+        },
+        {
+            name: "Farfetch\'d",
+            height: 2.07,
+            types: ['normal', 'flying']
+        },
+        {
+            name: "Jigglypuff",
+            height: 1.08,
+            types: ['normal', 'fairy']
+        },
+        {
+            name: "Oddish",
+            height: 1.08,
+            types: ['grass', 'poison']
+        }
+    ];
+
+    function getAll() {
+        return pokemonList;
+    }
+
+    function getSpecific(name) {
+        return pokemonList.filter(function(pokemon) {
+            return name === pokemon.name;
+        });
     }
-];
+
+    function add(newPokemon) {
+        if (typeof newPokemon === "object" && JSON.stringify(Object.keys(newPokemon)) === JSON.stringify(Object.keys(pokemonList[0]))) {
+            return pokemonList.push(newPokemon);
+        } else {
+            console.log(`The pokémon must be stored as an object with the keys: ${Object.keys(pokemonList[0])}.`);
+        }
+    }
+
+    return {
+        getAll: getAll,
+        add: add,
+        getSpecific: getSpecific
+    };
```

Comment:
This can be written with the short hand as well since the key-value pairs are the same.
```
return {
        getAll,
        add,
        getSpecific
    };
    ```